frontend/src/features/admin-form/create/payment/PaymentDrawer.tsx

```diff
@@ -267,11 +277,30 @@ export const PaymentInput = (): JSX.Element => {
   )
 }
 
-export const PaymentDrawer = ({
-  paymentsField,
+// Will be extended for stripe unconnected messages too
+const PaymentDisabledMessage = ({
+  isEncryptMode,
 }: {
-  paymentsField: FormPaymentsField
+  isEncryptMode: boolean
 }): JSX.Element | null => {
+  return !isEncryptMode ? (
```

Comment:
Would prefer to have final return to be the element that's intended to render if we're in the happy flow.

Example
```jsx
if (isEncryptMode) { return null }

return (<Box>...</Box>)
```

Reply:
On second thought, feels like this component shouldn't need to accept a `isEncryptMode` props. The parent should be the controller of whether this component should render since this component meant to show a `PaymentDisabledMessage` 🤔

Reply:
The reason for this is that being in email mode isn't the only cause for a `PaymentDisabledMessage`, #5849 which is what we were experimenting ytd is the other cause. 

Technically I could do the rendering and the switch inside PaymentDrawer itself but felt it will be cleaner if I abstracted it all out!

Reply:
in that case, is it necessary to have a separate component for this? I wonder if we can do this directly in the `PaymentDrawer` component instead:

```typescript
const PaymentDrawer = () => {
  // ...

  const isDisabledMessage = 
    isEmailMode 
    ? 'Payments are not available for email mode.' 
    : !form.payment_channel 
    ?  'Please add a payment channel before enabling payments'
    : ''

  const isDisabled = !!isDisabledMessage

  // ...

  return (
    // ...
    { isDisabledMessage && <Box> {isDisabledMessage} </Box>}
    // ...
  )
}

Reply:
Yeah thats possible too, I was considering between these two options, but in the end I felt separating it into its own function was 'cleaner' and potentially more extendable in the future (in case we every have more info/errors from logic etc)

Reply:
Hmm but if we have additional error messages, it will end up bloating the if else chain right?

In reference to how we did myinfo's inline messages (in FieldListDrawer), if we have additional checks in the future it might bloat the PaymentDrawer

Reply:
I don't feel strongly either way. If there is complexity, it has to go somewhere - either we expand the number of props taken by `PaymentDisabledMessage` or the expand the logic in `PaymentDrawer`. Maybe others can weigh in if they feel strongly.

Nonetheless circling back, i agree with Ken's initial nit which is to prefer having the happy path be the final return :P